src/components/Chat/AudioInput/index.js

```diff
@@ -0,0 +1,147 @@
+  const [recordingFinished, setRecorderFinished] = useState(false)
+  const timer = React.createRef()
+
+  // If timer has reached 60 seconds, it sets recordingFinished to true and this is activated
+  useEffect(() => {
+    if (recordingFinished) {
+      timer.current.stop()
+      recorder.stopRecording()
+      stream.stop()
+    }
+  }, [recordingFinished, recorder, stream, timer])
+
+  // This useEffect handles opening and closing audio stream
+  useEffect(() => {
+    if (isRecording) {
+      const openStream = async () => {
+        setStream(
+          await navigator.mediaDevices.getUserMedia({
+            audio: true,
+          })
+        )
+      }
+      if (!stream) {
+        openStream()
+      }
+    }
+    return () => {
+      if (stream) {
+        stream.stop()
+      }
+    }
+  }, [isRecording, stream])
+
+  // This useEffect handles creating and setting recorder after stream is set
+  useEffect(() => {
+    if (stream) {
+      setRecorder(
+        new RecordRTC(stream, {
+          type: 'audio',
+          mimeType: 'audio/wav',
+        })
+      )
+    }
+  }, [stream])
+
+  // This useEffect handles starting recorder after it is set.
+  useEffect(() => {
+    if (recorder) {
+      recorder.startRecording()
+    }
+  }, [recorder])
+
+  // If recording is finished (timer is at 60 secs) recorder is already stopped.
+  const endRecording = () => {
+    if (recordingFinished) {
+      const blob = recorder.getBlob()
+      const file = new File([blob], 'audio.wav')
+      handleSubmit(file)
+      closeModal()
+    } else if (recorder) {
+      recorder.stopRecording(() => {
+        const blob = recorder.getBlob()
+        const file = new File([blob], 'audio.wav')
+        handleSubmit(file)
+        closeModal()
+      })
+    } else {
+      closeModal()
+    }
+  }
+
+  // If recording is finished (timer is at 60 secs) recorder is already stopped.
+  const cancelRecording = () => {
+    if (recordingFinished) {
+      closeModal()
+    } else if (recorder) {
+      recorder.stopRecording(() => {
+        closeModal()
+      })
+    } else {
+      closeModal()
+    }
+  }
+
+  return (
+    <main className="audio-recording-content">
+      <div className="audio-recording-info">
+        {recorder && (
+          <Timer
+            ref={timer}
+            lastUnit="s"
+            checkpoints={[
+              { time: 60000, callback: () => setRecorderFinished(true) },
+            ]}
+          >
+            <div className="audio-timer">
+              <Timer.Seconds />
+            </div>
+          </Timer>
+        )}
+        <span
+          className={recordingFinished ? 'pulse-done' : 'pulse-recording'}
+        />
+      </div>
+      {recorder && (
+        <p className="audio-description-text">
+          {recordingFinished ? 'Äänitys valmis' : 'Äänitys käynnissä...'}
+        </p>
+      )}
+      {!recorder && (
+        <div className="audio-description-text">
+          <p>
```

Comment:
When I was testing this, I had blocked microphone from chrome, so app crashed right away. I added couple of conditions and info text to enable mic, if no recorder is present.